### DISCUSSION TITLE
Filter regions included in PDF report

### DISCUSSION THREAD

**Discussion by TerryA60** [code4recovery/pdf#85] [Ideas]
Our current PDF report includes all of eastern Missouri. We would like to filter this down to sub-regions to create local meeting guides. I am thinking of the ability to upload a list of regions to include in the report. A multi select box would be awkward because of the large number of regions to include or exclude. We would also like to be able to filter out online only meetings.

**Comment by anchovie91471** [comment]
PDF does not include online meetings. We are discussing, internally, the possibility of adding those to PDF, but you should not be getting any listed in a PDF from the service.

Let me do a bit of research on the filtering part and I'll get back to you.

I'm also going to convert this to a discussion until this turns into something actionable. 

Stay tuned!

**Comment by anchovie91471** [comment]
If you'll post your feed here, we can assist you in filtering the feed by sub-region. You can then use each feed to generate a separate PDF.

**Reply by TerryA60** [reply]
I am not sure of what you are asking. This is for the St. Louis Cenral Service office, serving eastern Missouri.
Are you asking for the list of included regions?

**Comment by anchovie91471** [comment]
No. Sorry. I was wanting the URL that you're pasting into the field, or, if you're clicking the link in the 12 Step Meeting List plugin, the ULR that shows up in the first field before you generate the PDF.

**Reply by TerryA60** [reply]
here is the URL https://aastl.org/wp-admin/edit.php?post_type=tsml_meeting

this seems right  https://aastl.org/wp-admin/admin-ajax.php?action=meetings
attached is an inclusion list of Regions for the PDF report. This is will be a monthly PDF print.
[RegionIncludeList.txt](https://github.com/user-attachments/files/17737490/RegionIncludeList.txt)

**Comment by anchovie91471** [comment]
@TerryA60 are you wanting a separate PDF for each of the Regions in your text file?

**Reply by TerryA60** [reply]
We want to produce a PDF that has only the "regions" that are in the St. Louis metropolitan area.
Currently our PDF includes all of eastern Missouri.
We can provide a list to include or exclude.
Terry

**Reply by joshreisner** [reply]
<img width="295" alt="Screenshot 2024-12-11 at 9 19 59 AM" src="https://github.com/user-attachments/assets/a2cd151c-fa58-4951-8bf8-8b5d3728d3a9" />

is it just these regions? one option is you could arrange the regions with a parent:

* St Louis
  * Central West End
  * Downtown
  * North City
  * South City
  * etc

**Reply by TerryA60** [reply]
Greetings. Attached is the list of Reg
[StLouisPDFinclusionList.csv](https://github.com/user-attachments/files/18186985/StLouisPDFinclusionList.csv)
ions we would like to include in our PDF Report for St. Louis.

**Comment by TerryA60** [comment]
We want St. Louis city plus the surrounding suburbs in St. Louis County in our PDFTerryOn Dec 11, 2024, at 11:22 AM, Josh Reisner ***@***.***> wrote:﻿
Screenshot.2024-12-11.at.9.19.59.AM.png (view on web)
is it just these regions? one option is you could arrange the regions with a parent:

St Louis

Central West End
Downtown
North City
South City
etc

**Comment by joshreisner** [comment]
@AASTL and i worked on this a bit today and here is a PDF we came up with: [4.25x11_day-region-grouped_2024-12-26_directory.pdf](https://github.com/user-attachments/files/18254937/4.25x11_day-region-grouped_2024-12-26_directory.pdf)

this address is what you can use to filter down to these regions - hopefully:
https://aastl.org/wp-admin/admin-ajax.php?action=meetings&region[]=affton&region[]=arnold&region[]=bellefontaine-neighbors&region[]=bel-nor&region[]=brentwood&region[]=bridgeton&region[]=byrnes-mill&region[]=chesterfield&region[]=clayton&region[]=cool-valley&region[]=crestwood&region[]=creve-coeur&region[]=desoto&region[]=ellisville&region[]=eureka&region[]=fenton&region[]=ferguson&region[]=florissant&region[]=franklin-county&region[]=frontenac&region[]=glendale&region[]=hazelwood&region[]=herculaneum&region[]=high-ridge&region[]=hillsboro&region[]=house-springs&region[]=imperial&region[]=jefferson-county-central&region[]=jefferson-county-east&region[]=jefferson-county-west&region[]=kirkwood&region[]=ladue&region[]=lemay&region[]=manchester&region[]=maplewood&region[]=maryland-heights&region[]=normandy&region[]=oakville&region[]=overland&region[]=pacific&region[]=pevely&region[]=south-stl-county&region[]=st.-louis-central-west-end&region[]=st.-louis-north-city&region[]=st.-louis-south-city&region[]=st.-louis-southwest-city&region[]=st.-louis-the-grove&region[]=st.-louis-tower-grove&region[]=town-and-country&region[]=university-city&region[]=valley-park&region[]=webster-groves&region[]=wellston&region[]=wildwoodo.code4recovery.org/wp-admin/admin-ajax.php?action=meetings

**Reply by joshreisner** [reply]
to add a new region you'll just need to add `&region[]=` and then the region slug (which you see in the URL when filtering your meeting list)

**Comment by anchovie91471** [comment]
@TerryA60 I believe this was solved by our latest release. It now has a filter for regions. Can you check and see if this provides the functionality you were looking for?
